### PR TITLE
feature: simplify audit wijziging converter class hierarchy

### PR DESCRIPTION
**Pull request by hsiliev**
Remove converter hierarchy

Solves: PZ-3261

### CODE REVIEW

File: src/main/kotlin/net/atos/zac/app/audit/converter/RESTHistorieRegelConverter.kt [hunk 1]
```diff
@@ -4,36 +4,46 @@
...
+class RESTHistorieRegelConverter @Inject constructor(
+    private val auditEnkelvoudigInformatieobjectConverter: AuditEnkelvoudigInformatieobjectConverter
+) {
 
     fun convert(auditTrail: List<AuditTrailRegel>): List<RESTHistorieRegel> =
         auditTrail.sortedByDescending { it.aanmaakdatum }
             .flatMap { convert(it) }
             .toList()
 
+    @Suppress("UNCHECKED_CAST")
```
Comment (hsiliev) [inline src/main/kotlin/net/atos/zac/app/audit/converter/RESTHistorieRegelConverter.kt:32]: Previously in https://github.com/infonl/dimpact-zaakafhandelcomponent/pull/1455/files#diff-990d49a95cecbfe522f927302aef0960c9986fdfb2672c2d784b50953b7038f0L13

File: src/main/kotlin/net/atos/zac/app/audit/converter/RESTHistorieRegelConverter.kt [hunk 2]
```diff
@@ -4,36 +4,46 @@
...
     private fun convert(auditTrailRegel: AuditTrailRegel): List<RESTHistorieRegel> =
```
Comment (edgarvonk) [inline src/main/kotlin/net/atos/zac/app/audit/converter/RESTHistorieRegelConverter.kt:33]: Could maybe make an extension function of this? Maybe for a future PR. :-)

**Review by edgarvonk — APPROVED** [review]
nice

File: src/main/kotlin/net/atos/zac/app/audit/converter/RESTHistorieRegelConverter.kt (hunk 2 above)
Comment (hsiliev) [reply src/main/kotlin/net/atos/zac/app/audit/converter/RESTHistorieRegelConverter.kt]: Done. It definitely reads better. 

From a single audit trail event we produce several, so flatMap is still needed it seems.

**Pull request merged**